### DISCUSSION TITLE
feat: GroqCloud integration

### DISCUSSION THREAD

**Discussion by 0x7CFE** [open-webui/open-webui#2307] [General]
**Is your feature request related to a problem? Please describe.**
GroqCloud now provides free access to its inference cluster which is too fast to ignore. Also it supports LLama3-**70b** out of the box with speeds up to **300 tokens/s**! This is more than enough for a curious person's use.

**Describe the solution you'd like**
I'd like to be able to use Open Web UI to act as an interface to GroqCloud to keep all my chats in one place.

**Describe alternatives you've considered**
- Use Groq's own UI. Inconvenient for free users, and feature poor.
- Use Telegram chatbots. Work fine but again, I'd like to have it all in one place.
- Use own ollama instance. Speed is considerably lower.

**Additional context**
https://groq.com/
https://console.groq.com/playground
https://console.groq.com/docs/quickstart

```bash
curl "https://api.groq.com/openai/v1/chat/completions" \
  -X POST \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer ${GROQ_API_KEY}" \
  -d '{
         "messages": [
           {
             "role": "user",
             "content": "Explain the importance of fast language models"
           },
         ],
         "model": "llama3-70b-8192",
         "temperature": 1,
         "max_tokens": 1024,
         "top_p": 1,
         "stream": true,
         "stop": null
       }'
  
  ```

**Comment by gtasteve** [comment]
Add the Groq api to open-webui to use all their LLMs
https://github.com/open-webui/open-webui/discussions/974#discussioncomment-9632355
